Makefile

```diff
@@ -81,25 +81,30 @@ ROCKSDB_SYS_PORTABLE=0
 RUST_TEST_THREADS ?= 2
 endif
 
-# Disable SSE on ARM
-ifeq ($(shell uname -p),aarch64)
-ROCKSDB_SYS_SSE=0
+# Enable sse4.2 by default on x86 and x64
+ifeq ($(shell uname -m),x86_64)
```

Comment:
How about test by `cc -msse3 -dM -E - </dev/null|grep SSE`?

Reply:
Looks like this will be very similar to how RocksDB enables SSE intelligently in https://github.com/tikv/rocksdb/blob/ce0ea61aecd53ac02a0e7da2b166961c7d98650a/CMakeLists.txt#L333 (by looking at the instruction set availability instead of the platform). There is a check SSE script doing similar things, I'm not sure whether this is fine.

Reply:
This is too ugly. How about utilizing `rust -vV`? If there is 86 in the host line, we can force enabling SSE.

Reply:
Good idea!